vividus-studio-server/vividus-studio-plugin/src/main/java/org/vividus/studio/plugin/document/TextDocumentEditor.java

```diff
@@ -0,0 +1,172 @@
+/*-
+ * *
+ * *
+ * Copyright (C) 2020 - 2021 the original author or authors.
+ * *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * *
+ */
+
+package org.vividus.studio.plugin.document;
+
+import static org.apache.commons.lang3.StringUtils.splitByWholeSeparatorPreserveAllTokens;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.lsp4j.DidChangeTextDocumentParams;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentContentChangeEvent;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class TextDocumentEditor implements TextDocumentEventListener
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(TextDocumentEditor.class);
+    private final Map<String, List<String>> textDocuments = new HashMap<>();
```

Comment:
I propose to clean-up during onClose event

Reply:
fixed